### ISSUE TITLE
Proposal: Option to log all requests

### ISSUE THREAD

**Issue by MarvinJWendt** [pocketbase/pocketbase#947]
Since we are using PocketBase as a framework, we also provide our frontend code via the same router as the PocketBase API (echo). The Logs in the UI only show requests made to the API. It would be useful to have all request logged, so that we can use the UI to also filter for errors that occurred outside the API requests.

We are currently using this as a workaround:

```go
    app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
		e.Router.Use(func(next echo.HandlerFunc) echo.HandlerFunc {
			return func(c echo.Context) error {
				c.Response().Before(func() {
					go func() {
						_ = app.LogsDao().Save(&models.Request{
							BaseModel: models.BaseModel{},
							Url:       c.Request().URL.String(),
							Method:    c.Request().Method,
							Status:    c.Response().Status,
							Auth:      "",
							UserIp:    c.RealIP(),
							RemoteIp:  "",
							Referer:   c.Request().Referer(),
							UserAgent: c.Request().UserAgent(),
							Meta:      map[string]any{},
						})
					}()
				})
				return next(c)
			}
		})

    return nil
    })
```

While this workaround does its job, I think an option in PocketBase would be cool to enable logging of all requests.

The background of this is, that it would be logical that PocketBase can log all requests, if it also serves the frontend, to detect errors in the whole application, using a single UI.

**Comment by ganigeorgiev** [comment]
@MarvinJWendt You don't have to manually save the log request. The internal activity logger middleware is already exposed as  `apis.ActivityLogger(app)` and you can attach it to your custom endpoints like this:

```go
app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
	// enable activity logger for custom api route
	e.Router.AddRoute(echo.Route{
		Method: http.MethodGet,
		Path:   "/api/custom/hello",
		Handler: func(c echo.Context) error {
			return c.String(http.StatusOK, "Hello world!")
		},
		Middlewares: []echo.MiddlewareFunc{
			apis.ActivityLogger(app),
		},
	})

	// enable activity logger for the file serving endpoint
	e.Router.GET("/*", apis.StaticDirectoryHandler(os.DirFS("./public"), true), apis.ActivityLogger(app))

	return nil
})
```

I'll make sure to add an example in the new docs that will be released with the upcoming v0.8.0.

**Comment by denisgurev** [comment]
Thank you @ganigeorgiev. `ActivityLogger` should definitely be documented somewhere. I was just gonna post a question asking how to do it and found this issue 👍
